ompi/mca/coll/base/coll_base_util.c

```diff
@@ -297,11 +298,232 @@ int ompi_coll_base_retain_datatypes_w( ompi_request_t *req,
+        if ('#' == trash) {
+            skiptonewline (fptr, fileline);
+        }
+    } while (1);
+}
+
+int ompi_coll_base_file_getnext_size_t(FILE *fptr, int *fileline, size_t* val)
+{
+    char trash;
+    int rc;
+
+    do {
+        rc = fscanf(fptr, "%" PRIsize_t, val);
+        if (rc == EOF) {
+            return -1;
+        }
+        if (1 == rc) {
+            return 0;
+        }
+        /* in all other cases, skip to the end of the token */
+        rc = fread(&trash, sizeof(char), 1, fptr);
+        if (rc == EOF) {
+            return -1;
+        }
+        if ('\n' == trash) (*fileline)++;
+        if ('#' == trash) {
+            skiptonewline (fptr, fileline);
+        }
+    } while (1);
+}
+
+int mca_coll_base_name_to_colltype(const char* name)
+{
+    if( 0 == strncmp(name, "neighbor_all", 12) ) {
+        if( 't' != name[12] ) {
+            if( 0 == strcmp(name+12, "allgatherv") )
+                return NEIGHBOR_ALLGATHERV;
+            if( 0 == strcmp(name+12, "allgather") )
+                return NEIGHBOR_ALLGATHER;
+        } else {
+            if( 0 == strcmp(name+12, "alltoallv") )
+                return NEIGHBOR_ALLTOALLV;
+            if( 0 == strcmp(name+12, "alltoallw") )
+                return NEIGHBOR_ALLTOALLW;
+            if( 0 == strcmp(name+12, "alltoall") )
+                return NEIGHBOR_ALLTOALL;
+        }
+        return -1;
+    }
+    if( 0 == strncmp(name, "all", 3) ) {
+        if( 't' != name[3] ) {
+            if( 0 == strcmp(name+3, "gatherv") )
+                return ALLGATHERV;
+            if( 0 == strcmp(name+3, "gather") )
+                return ALLGATHER;
+            if( 0 == strcmp(name+3, "reduce") )
+                return ALLREDUCE;
+        } else {
+            if( 0 == strcmp(name+3, "toallv") )
+                return ALLTOALLV;
+            if( 0 == strcmp(name+3, "toallw") )
+                return ALLTOALLW;
+            if( 0 == strcmp(name+3, "toall") )
+                return ALLTOALL;
+        }
+        return -1;
+    }
+    if( 'r' >= name[0] ) {
+        if( 0 == strcmp(name, "barrier") )
+            return BARRIER;
+        if( 0 == strcmp(name, "bcast") )
+            return BCAST;
+        if( 0 == strcmp(name, "exscan") )
+            return EXSCAN;
+        if( 0 == strcmp(name, "gatherv") )
+            return GATHERV;
+        if( 0 == strcmp(name, "gather") )
+            return GATHER;
+    }
+    if( 's' >= name[0] ) {
+        if( 0 == strcmp(name, "reduce_scatter_block") )
+            return REDUCESCATTERBLOCK;
+        if( 0 == strcmp(name, "reduce_scatter") )
+            return REDUCESCATTER;
+        if( 0 == strcmp(name, "reduce") )
+            return REDUCE;
+    }
+    if( 0 == strcmp(name, "scan") )
+        return SCAN;
+    if( 0 == strcmp(name, "scatterv") )
+        return SCATTERV;
+    if( 0 == strcmp(name, "scatter") )
+        return SCATTER;
+    return -1;
+}
+
+const char* mca_coll_base_colltype_to_str(int collid)
```

Comment:
I suggest using a conversion table to simplify this lengthy switch statement

Reply:
Because the switch case values are monotonically increasing this format is automatically optimized as a table by most compilers nowadays.

Reply:
The compilers should - but benchmarking at least for gcc a conversion table is faster... (see https://github.com/open-mpi/ompi/pull/7735#discussion_r462899120)